### PR TITLE
Is721/jupyter persistency

### LINKED ISSUE #616: Director fails while retrieving services
``aiohttp.web_exceptions.HTTPInternalServerError: Expecting property name enclosed in double quotes: line 1 column 112 (char 111)``

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- jupyter-base-notebook:
  - [x] add API to push/pull state of Jupyter node from/to S3
  - [x] notebook automatically pulls state from S3 before starting if possible
  - [x] notebook automatically pushes state when it saves its state (post-save-hook)
  - [x] updated to 2.9.0

- all-jupyter based analysis services:
  - [x] update base image to 2.9.0
  - [x] add auto push/pull mechanism

- platform:
  - [x] director passes the project ID to the running nodes for allow push/pull in S3
  - [x] webserver requires and passes the project ID to the director when starting services
  - [x] enhanced simcore_sdk package to have a generic way of pushing/pulling a file or folder to simcore S3
  - [x] update front-end to correctly call webserver (thanks @odeimaiz)

Bonuses:
- [x] remove waffle from repository
- [x] updated makefile to latest conventions
- [x] adapted devel mode to use latest staging versions of storage, apihub
- [x] upgraded sqlalchemy to 1.3.x as of security issue with <1.3.0 (#725)
- [x] upgraded urllib3 to 1.24.2 as of security issue with <=1.24.1 (#725)
- [x] refactoring of docker-compose files in jupyter for easier reading
- [x] fixes #616 
## Related issue number
#721 
#649 
#616 
jupyter-persistency

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet

### CODE REVIEW

**Comment by sanderegg** [comment]
SIMCORE_NODE_APP_STATE_PATH:
- name still not totally clear apparently ;)
- is a node-defined location containing the state of the node
- is totally independent of project_id/node_id, it's local to the node

**Comment by odeimaiz** [comment]
Happy with the frontend part

File: packages/simcore-sdk/tests/node_data/test_data_manager.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    # create a folder to compress from
+    test_folder = Path(tmpdir) / "test_folder"
+    test_folder.mkdir()
+    assert test_folder.exists()
+    # create a folder to compress in
+    test_compression_folder = Path(tmpdir) / "test_compression_folder"
+    test_compression_folder.mkdir()
+    assert test_compression_folder.exists()
+
+    mock_filemanager = mocker.patch('simcore_sdk.node_data.data_manager.filemanager', spec=True)
```
Comment (pcrespov) [inline packages/simcore-sdk/tests/node_data/test_data_manager.py]: 👍 very nice!

File: services/director/temp_generate_openapi.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+if [ -z "$apihub_specs_dir" ]; then
+    echo "please define an apihub specs directory..."
+    usage
+    exit 1
+fi
+
+docker run \
+    -v $apihub_specs_dir:/input \
+    -v ${PWD}/src/simcore_service_director/oas3/v0:/output \
+    itisfoundation/oas_resolver \
```
Comment (pcrespov) [inline services/director/temp_generate_openapi.sh]: all tooling containers should be defined in https://github.com/ITISFoundation/dockerfiles 
should move there also itisfoundation/oas_resolver be

File: api/specs/director/v0/openapi.yaml [hunk 3]
```diff
@@ -138,6 +138,7 @@ paths:
       operationId: running_interactive_services_post
       parameters:
         - $ref: '#/components/parameters/UserId'
+        - $ref: '#/components/parameters/ProjectId'
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply api/specs/director/v0/openapi.yaml]: well, I heard about this yesterday concerning the API... so I kept the current naming scheme... anyway the API needs a refresh.

File: services/director/temp_generate_openapi.sh (hunk 2 above)
Comment (sanderegg) [reply services/director/temp_generate_openapi.sh]: yep, but this will disappear once the director is re-made using the cookiecutter

**Comment by sanderegg** [comment]
@pcrespov we need to sync the formatting features you use in vscode...

**Comment by pcrespov** [comment]
@sanderegg Upss... sorry for that. I use the ones in the template

File: api/specs/webserver/v0/openapi.yaml [hunk 4]
```diff
@@ -309,6 +310,16 @@ components:
         # format: uuid
         example: 123e4567-e89b-12d3-a456-426655440000
 
+    ProjectIdQuery:
+      in: query
+      name: project_id
+      required: true
+      description: starts the pipeline
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply api/specs/webserver/v0/openapi.yaml]: done!

**Comment by sanderegg** [comment]
@pcrespov no worries, I do not use anything in particular for formatting, I tried it once to have it automatic but then removed it cause I did not like some parts of what it was doing...

**Pull request merged**